lib/CachingOperation.ts

```diff
@@ -10,4 +12,27 @@ export class CachingOperation<LoadedValue> extends AbstractFlatOperation<LoadedV
       })
     }
   }
+  public setAsync(key: string, newValueFn: () => Promise<LoadedValue>): Promise<LoadedValue> {
+    const existingLoad = this.runningSetAsyncLoads.get(key)
+    if (existingLoad) {
+      return existingLoad
+    }
+
+    const resolverAndSetPromise = new Promise<LoadedValue>((resolve, reject) => {
```

Comment:
why wrap it into a new promise and not just reuse the one created from `cacheValueResolver`?